internal/ethapi/api.go

```diff
@@ -237,17 +238,17 @@ func (s *PrivateAccountAPI) ImportRawKey(privkey string, password string) (commo
 // UnlockAccount will unlock the account associated with the given address with
 // the given password for duration seconds. If duration is nil it will use a
 // default of 300 seconds. It returns an indication if the account was unlocked.
-func (s *PrivateAccountAPI) UnlockAccount(addr common.Address, password string, duration *hexutil.Uint) (bool, error) {
+func (s *PrivateAccountAPI) UnlockAccount(addr common.Address, password string, duration *int64) (bool, error) {
 	var d time.Duration
 	if duration == nil {
 		d = 300 * time.Second
+	} else if *duration < 0 {
```

Comment:
Hmm. I prefer type system checks over manual ones (less code, more obvious, etc). Maybe we should rather fix the unmarshaller so that it can spit out appropriate error messages for negative uints.

Also I don't think it's common for people to pass in negative numbers, so user friendly error messages may be a bit overkill for a scenario that will probably not happen.

@obscuren @Arachnid Can one of you decide this? I can see it go both ways, would rather go with the one preferred by the majority instead of forcing my right :)